test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -345,17 +345,13 @@ func setupCRD(f *framework.Framework, schema []byte, groupSuffix string, version
 	}
 
 	crd, err := crd.CreateMultiVersionTestCRD(f, group, func(crd *v1beta1.CustomResourceDefinition) {
-		apiVersions := []v1beta1.CustomResourceDefinitionVersion{}
-		for _, version := range versions {
-			v := v1beta1.CustomResourceDefinitionVersion{
+		for i, version := range versions {
+			crd.Spec.Versions = append(crd.Spec.Versions, v1beta1.CustomResourceDefinitionVersion{
 				Name:    version,
 				Served:  true,
-				Storage: false,
-			}
-			apiVersions = append(apiVersions, v)
+				Storage: i == 0,
+			})
 		}
-		apiVersions[0].Storage = true
-
```

Comment:
would it be clearer to overwrite `crd.Spec.Version = apiVersions` here? it's unclear to a caller whether crd.Spec.Versions already has versions in it that we are appending to

Reply:
done